jupyterhub_moss/spawner.py

```diff
@@ -351,70 +342,103 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
 
         self.__validate_options(options)
 
-        partition = options["partition"]
+        partition_environments = tuple(
+            self.partitions[options["partition"]]["jupyter_environments"].values()
+        )
+        if "environment_path" not in options:
+            # Set path to use from first environment for the current partition
+            options["environment_path"] = partition_environments[0]["path"]
```

Comment:
Those updates of `options` can be kept here.

I split updates or `options` between the one relying on "static" info and the one relying on "dynamic" info, but not sure it is really needed?